File: Data Structures & Algorithms/search-2d-matrix/submission-0.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        lst = []

        for i in matrix:
            lst += i
        print(lst)
        
        left = 0
        right = len(lst) - 1

        while left <= right:
            middle = left + (right - left) // 2
            if lst[middle] == target:
                return True
            elif lst[middle] > target:
                right = middle - 1
            else:
                left = middle + 1
                        
        return False

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-0.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        Set<Integer> window = new HashSet<>();
        int left = 0;
        int sum = 0;
        int minSize = Integer.MAX_VALUE;

        for (int right = 0; right < nums.length; right++){
            sum += nums[right];
            if (sum >= target){
                while (sum >= target){
                    minSize = Math.min(minSize, right - left + 1);
                    sum -= nums[left];
                    left++;
                }
            }
        }
        if (minSize != Integer.MAX_VALUE){
            return minSize;
        }
        return 0;
    }
}